Project 5 For John Beverly

```diff
@@ -59,345 +59,388 @@ Group 2 contains Emanuel, Katie, Josh V (commonly know as V by group member), Ca
 Please note we DID check this against SHACL playground :) 
 enjoy, if possible.
 
-
-1st constraints begins below
-
-@prefix ex: <http://example.org/> .
-
+1.	Constraint for Asymmetry 
 @prefix sh: <http://www.w3.org/ns/shacl#> .
 
-@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix ex: <http://example.com/> .
 
 @prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-@prefix schema: <http://schema.org/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
-ex:RegistrationShape
+ex:AsymmetryShape
 
 a sh:NodeShape ;
 
-sh:targetClass ex:Registration ;
+sh:targetClass ex:relation ;
 
 sh:property [    
 
-sh:path ex:lineRegistration ;    
+sh:path rdf:value ;    
 
-sh:order ex:lineRegistrationOrder ;    
+sh:minCount 1 ;    
 
-sh:node ex:RegistrationShape ; 
+sh:maxCount 1 ;    
+
+sh:datatype xsd:string ; 
 
 ] .
 
+2.	Constraint for Symmetry 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
 
-ex:lineRegistrationOrder
+@prefix ex: <http://example.com/> .
 
-a sh:Order ;
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-sh:datatype xsd:integer ;
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
-sh:path ex:addedAt ;
 
-sh:ascending true ;
+ex:SymmetryShape
 
-sh:uniqueMembers true .
+a sh:NodeShape ;
 
+sh:targetClass ex:relation ;
 
-ex:RegistrationDateShape
-  
-  a sh:NodeShape ;
-  
-  sh:targetClass ex:RegistrationDate ;
-  
-  sh:property [    
-  
-  sh:path ex:RegistrationDate ;    
-  
-  sh:datatype xsd:string ;   
- 
- Sh:description
- 
- "US Date - format - MM/DD/YYYY" ;     
+sh:property [    
 
-sh:minCount 1 ;   
+sh:path rdf:value ;    
 
-sh:maxCount 1 ;   
+sh:minCount 1 ;    
 
-sh:name "RegistrationDate" ;    
+sh:maxCount 1 ;    
 
-sh:pattern "[1-12]/[1-31]/[1990-2020]”" ;  
+sh:datatype xsd:string 
 
 ] .
 
+3.	Constraint Symmetry with a check things not being symmetric 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
 
+@prefix ex: <http://example.com/> .
 
-ex:lineRegistrationDateOrder
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 
-a sh:Order ;
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
 
-sh:datatype xsd:integer ;
+ex:SymmetryShape
 
-sh:path ex:RegistrationAt ;
+a sh:NodeShape ;
 
-sh:ascending true ;
+sh:targetclass ex:relation ;
 
-sh:uniqueMembers true .
+sh:property [    
 
+sh:path rdf:value ;    
 
-ex:RegistrationDateANDlineRegistrationOrder
+sh:minCount 2 ;    
 
-a sh:NodeShape ;
+sh:datatype xsd:string ;    
 
-sh:and (ex:RegistrationDateShape ex:lineRegistrationDateOrder) ;
+sh:nodeKind sh:Literal ;    
 
-sh:property [    
+sh:message "The relation does not have symmetric values"  
 
-sh:path rdf:type ;    
+] ;
 
-sh:datatype xsd:integer ;   
+sh:property [    
 
-sh:node ex:ValidationResult ;   
+sh:path [ 
 
-sh:resultSeverity sh:Violation ;   
+sh:inversePath rdf:value 
```

Comment:
I am not sure this is the right syntax for inversepath (see slide 38 from John's lecture)